Release notes for edge-21.4.4 (#6056)
linkerd/linkerd2: CHANGES.md

# Changes

## edge-21.4.4

This edge release further consolidates the control plane by removing the
`linkerd-controller` deployment and moving the `sp-validator` container into the
`destination` deployment.

> Annotation inheritance has been added so that +all* Linkerd annotations

Reviewer: Not sure the emphasis is needed here:
```suggestion
Annotation inheritance has been added so that all Linkerd annotations
```

> Annotation inheritance has been added so that +all* Linkerd annotations

Reviewer 2: ```suggestion
Annotation inheritance has been added so that *all* Linkerd annotations
```

> on a `Namespace` resource will be inherited by pods within that namespace.

Reviewer: Nit: maybe try to reduce the amount of "code" blocks in summary text
```suggestion
on a namespace will be inherited by pods within that namespace.
```

In addition, the `config.linkerd.io/proxy-await` annotation has been added which
enables the [linkerd-await](https://github.com/olix0r/linkerd-await)

> functionality, simplifying the implementation of the await behavior.

Reviewer: We should mention this is enabled by default. Users can disable it by setting this annotation to `disabled`.

Some of the `linkerd check` functionality has been updated. The command
ensures that annotations and labels are properly located in the YAML and

> displays hinit anchors for core health checks.

Reviewer: ```suggestion
displays hint anchors for core health checks.
```

> displays hinit anchors for core health checks.

Reviewer 2: What are you referring to here about the hint anchors? There was a bug related to anchors, but that got fixed in 2.10.1

Author: That note came from #6023 

If it's the same thing that is in 2.10.1 and it doesn't need to be included, I'll remove it

Reviewer 2: Yep, that made it into 2.10.1

> Finally, the nginx container has been removed from the multicluster gateway pod,

Reviewer: Nit: capitalize extension name
```suggestion
Finally, the nginx container has been removed from the Multicluster gateway pod,
```

which will impact upgrades. Please see the note below.

> **Upgrade note:** When the multicluster extension is updated in both of the

Reviewer: ```suggestion
**Upgrade note:** When the Multicluster extension is updated in both of the
```

source and target clusters there won't be any downtime because this change only
affects the readiness probe. The multicluster links must be re-generated with

> the `linkerd mc link` command and the linkerd multicluster gateways will show

Reviewer 2: ```suggestion
the `linkerd mc link` command and the `linkerd multicluster gateways` command will show
```

the target cluster as not alive until the `linkerd mc link` command is re-run,
however that shouldn't affect existing endpoints pointing to the target cluster.

* Added proxy checks for core control plane and extension pods
* Added support for awaiting proxy readiness using an annotation
* Added namespace annotation inheritance to pods
* Removed the `linkerd-controller` pod
* Moved `sp-validator` container into the `destination` deployment
* Added check verifying that labels and annotations are not mixed up
  (thanks @szymongib@)

> * Enabled support for extra initContainers to a daemonset (thanks @mhulscher!)

Reviewer 2: ```suggestion
* Enabled support for extra initContainers to the linkerd2-cni daemonset (thanks @mhulscher!)
```

# Changes

## edge-21.4.4

This edge release further consolidates the control plane by removing the
linkerd-controller deployment and moving the sp-validator container into the
destination deployment.

Annotation inheritance has been added so that all Linkerd annotations
on a namespace resource will be inherited by pods within that namespace.
In addition, the `config.linkerd.io/proxy-await` annotation has been added which

> enables the [linkerd-await](https://github.com/olix0r/linkerd-await)

Reviewer 2: ```suggestion
enables the [linkerd-await](https://github.com/linkerd/linkerd-await)
```

Some of the `linkerd check` functionality has been updated. The command
ensures that annotations and labels are properly located in the YAML and adds
proxy checks for the control plane and extension pods.

Finally, the nginx container has been removed from the Multicluster gateway pod,
which will impact upgrades. Please see the note below.

**Upgrade note:** When the Multicluster extension is updated in both of the
source and target clusters there won't be any downtime because this change only
affects the readiness probe. The multicluster links must be re-generated with
the `linkerd mc link` command and the `linkerd mc gateways` will show
the target cluster as not alive until the `linkerd mc link` command is re-run,
however that shouldn't affect existing endpoints pointing to the target cluster.

* Added proxy checks for core control plane and extension pods
* Added support for awaiting proxy readiness using an annotation
* Added namespace annotation inheritance to pods
* Removed the linkerd-controller pod
* Moved sp-validator container into the destination deployment
* Added check verifying that labels and annotations are not mixed up

>   (thanks @szymongib@)

Reviewer 3: ```suggestion
  (thanks @szymongib)
```
